features/lwipstack/LWIPStack.cpp

```diff
@@ -493,7 +494,16 @@ nsapi_size_or_error_t LWIP::socket_sendto(nsapi_socket_t handle, const SocketAdd
     if (!convert_mbed_addr_to_lwip(&ip_addr, &addr)) {
         return NSAPI_ERROR_PARAMETER;
     }
-
+    struct netif *netif_ = netif_get_by_index(s->conn->pcb.ip->netif_idx);
+    if (!netif_) {
+        netif_ = &default_interface->netif;
+    }
+    if (netif_) {
+        if ((addr.version == NSAPI_IPv4 && !get_ipv4_addr(netif_)) ||
```

Comment:
BTW, I didn't consider V4/V6-only compilation - it looks like these `get_ipvx_addr` aren't defined if `LWIP_IPVX` isn't enabled. You should probably fiddle them to have them defined and return constant NULL in that case - will be neater than adding ifdefs here.

Reply:
Changed  get_ipvx_addr  LWIP_IPV4/6